### DISCUSSION TITLE
[BUG] Remove-PnPTenantSite -> This site belongs to a Microsoft 365 group. To delete the site, you must delete the group.

### DISCUSSION THREAD

**Discussion by LockTar** [pnp/powershell#3760] [General]
### Reporting an Issue or Missing Feature
I'm trying to immediately remove a SharePoint site permanently. So, a M365 group with a site that is deleted and not placed in de recycle bin or remove directly from recycle bin.

### Expected behavior 
Site is deleted and not placed in the recycle bin. 

### Actual behavior 
When running `Remove-PnPTenantSite -Url $siteUrl -Force -SkipRecycleBin` I get the error message `This site belongs to a Microsoft 365 group. To delete the site, you must delete the group.`.

### Steps to reproduce behavior

1. Create a new M365 group in the Azure portal called 'testgroup1'. Result is a group and a sharepoint site.
2. Call the CmdLet and you get the error message.

I also tested that when you delete the group the site is automatically deleted after around 15 minutes. But then, what is the point of the cmdlet then? I can't use it? Maybe it can be used with older sites that aren't connected to a group??? We need to delete the site immediately. 

### What is the version of the Cmdlet module you are running?
Latest 2.3.0

### Which operating system/environment are you running PnP PowerShell on?
- [x] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [x] Azure Functions
- [ ] Other : please specify

### Complete script
This script (when it's working) will be placed in an Azure Function with certificate authentication. 

```PS1
$filePath = "."
$AzAppId = "ClientId of app registration here"
$certFileName = "PnPProvisioningDemo.pfx"
$tenant = "tenant.onmicrosoft.com"

$url = "https://tenant.sharepoint.com"

Import-Module ("{0}/{1}" -f $filePath,"\PnP.PowerShell\2.3.0\PnP.PowerShell.psd1")
Connect-PnPOnline -ClientId $AzAppId -Tenant $tenant -CertificatePath ("{0}\{1}" -f $filePath,$certFileName) -Url $url

$siteName = "testralph10"
$siteUrl = "$($url)/sites/$($siteName)"

# Get active site by URL
$site = Get-PnPTenantSite -Url $siteUrl 
# Property GroupId contains the ID of the connected M365 group

if ($null -eq $site) {
    Write-Output "Site not found"
}
else {
    Write-Output "Found site Url: $($site.Url) and name: $($site.Title) and group ID: $($site.GroupId)"

    $group = Get-PnPMicrosoft365Group -Identity $site.GroupId

    if ($null -eq $group) {
        Write-Output "Group not found"
    }
    else {
        Write-Output "Found group ID: $($group.Id)"
        
        Write-Output "Remove group ID: $($group.Id)"
        Remove-PnPMicrosoft365Group -Identity $group.Id
        
        # Wait for 5 seconds to let the group be deleted
        Start-Sleep -Seconds 5

        # Cannot remove deleted groups because of permissions...? Which permissions?
        # Write-Output "Remove group ID: $($group.Id) from deleted groups"
        # Remove-PnPDeletedMicrosoft365Group -Identity $group.Id
    }

    # Delete site by URL
    Write-Output "Remove site with URL: '$($siteUrl)' and skip recycle bin"
    Remove-PnPTenantSite -Url $siteUrl -Force -SkipRecycleBin # Failes because of 'This site belongs to a Microsoft 365 group. To delete the site, you must delete the group.'
}

# Get all deleted sites
$allDeletedSiteUrls = Get-PnPTenantDeletedSite | Select-Object Url

if($allDeletedSiteUrls.Count -eq 0) {
    Write-Output "No deleted sites found"
}
else {
    Write-Output "Found $($allDeletedSiteUrls.Count) deleted sites. Removing..."

    foreach ($deletedSiteUrl in $allDeletedSiteUrls) {
        Write-Output "Remove deleted site with URL: $($deletedSiteUrl.Url)"
        Remove-PnPTenantDeletedSite -Identity $deletedSiteUrl.Url -Force            
    }    
}
```

**Comment by a1mery** [comment]
Hi,

I think this is by design.
As you said you can use this cmdlet with sites not connected to a group (such as Communication sites).

**Comment by LockTar** [comment]
But do we know this for sure? Is there another way to do this? Waiting until a site is deleted is not really a solution for us.

**Comment by LockTar** [comment]
I did another test.
I created again a new group (and so a new site). 

When you go to the SharePoint Admin Center -> Active sites and there delete the site, the site is deleted immediately. You first get the following confirmation `This site belongs to a Microsoft 365 group. Deleting the site will delete the group and all its resources, including the Outlook mailbox and calendar, and any Teams channels. You will have 30 days to restore the group.`.

The result is that the site is placed in the "Deleted sites" section and the M365 group is placed in the "Deleted groups" blade in the Azure portal.
After that, I can delete the deleted site **without** first delete the deleted group. When using PowerShell you first had to delete the deleted group before you could permanently delete the deleted site.

So, the SharePoint admin center is doing different commands then the PowerShell CmdLets. 

Maybe the SharePoint site first disconnects the group??? Is there a CmdLet for? 
Someone an idea?

**Reply by gautamdsheth** [reply]
You can also use `Remove-PnPMicrosoft365Group` cmdlet, maybe that helps.
As mentioned by @a1mery , this cmdlet is for deleting site collections. For team sites, this is server side error which might occur since Group and Sites are connected. If you create a communication site or some other type of site, then this will work fine.

**Reply by LockTar** [reply]
I tried the Remove-PnPMicrosoft365Group cmdlet. That is removing the group and indeed after 15 minutes the site but that is in our case to long.

**Comment by NoralK** [comment]
If there is a retention policy the site would not be deleted.

**Reply by LockTar** [reply]
The site is deleted but only after x minutes. So that is not working in our case.
